import "iw/iw.proto";

package labelprop_eval2;

message Params {
  required uint32 num_trials = 1;
  required float frac_aux_images = 2;
  required uint32 num_training_images = 3;
  enum SupervisionType {
    DETECTION = 1;
    LOCALIZATION = 2;
  };
  required SupervisionType supervision = 4;

  required double min_score_threshold = 5 [default = 2e-5]; // if a score is less than this, we don't try to guess
  required double uniqueness_threshold = 6 [default = 1.5];//  number of orders of magnitude by which the first best score must exceed the second best score
}

message Result {
  required Params params = 1;
  repeated uint32 label_ids = 2;
  repeated string label_names = 3;

  required iw.Gaussian precision = 4;
  required iw.Gaussian recall = 5;

  required iw.GaussianVector object_precision = 6;  // length: N
  required iw.GaussianVector object_recall = 7; // length: N

  required iw.GaussianMatrix confusion_matrix = 8; // dim: N x N
  required iw.ConfusionMatrixItemFreq confusion_matrix_item_freq = 9; // dim: N x N
}

